src/configuration.c

```diff
@@ -83,41 +84,48 @@ void set_config_file(char *filename)
 
 bool read_config_from_file(char *config_file, struct configuration *config)
 {
-        GError *error = NULL;
-        GKeyFileFlags flags = G_KEY_FILE_NONE;
-
         if (keyfile != NULL) {
-                g_key_file_free(keyfile);
+		nc_hashmap_free(keyfile);
         }
 
-        keyfile = g_key_file_new();
-
-        if (!g_key_file_load_from_file(keyfile, config_file, flags, &error)) {
+	keyfile = nc_ini_file_parse(config_file);
+        if (!keyfile) {
 #ifdef DEBUG
                 fprintf(stderr, "ERR: Failed to read config file: %s\n",
                         error->message);
 #endif
                 return false;
         } else {
                 for (int i = CONF_STR_MIN + 1; i < CONF_STR_MAX; i++) {
-                        config->strValues[i] = g_key_file_get_string(keyfile,
-                                                                     "settings", config_key_str[i], &error);
-                        if (error != NULL) {
-                                return false;
-                        }
+                        config->strValues[i] = nc_hashmap_get(nc_hashmap_get(keyfile,  "settings"), config_key_str[i]);
+			fprintf(stderr, "Looking for %s found %s \n", config_key_str[i], config->strValues[i]);
+			if (config->strValues[i] == NULL) {
+				return false;
+			}
                 }
                 for (int i = CONF_INT_MIN + 1; i < CONF_INT_MAX; i++) {
-                        config->intValues[i] = g_key_file_get_int64(keyfile,
-                                                                    "settings", config_key_int[i], &error);
-                        if (error != NULL) {
-                                return false;
+			char *ptr;
+			ptr = nc_hashmap_get(nc_hashmap_get(keyfile,  "settings"), config_key_int[i]);
+			if (ptr) {
+	                        config->intValues[i] = strtoll(ptr, NULL, 10);
+			} else {
+				return false;
                         }
                 }
 
                 for (int i = CONF_BOOL_MIN + 1; i < CONF_BOOL_MAX; i++) {
-                        config->boolValues[i] = g_key_file_get_boolean(keyfile,
-                                                                       "settings", config_key_bool[i], &error);
-                        if (error != NULL) {
+			char *ptr;
+			ptr = nc_hashmap_get(nc_hashmap_get(keyfile,  "settings"),  config_key_bool[i]);
+			config->boolValues[i] = false;
+
+			if (ptr) {			
+				if (strcasecmp(ptr, "TRUE")==0) {
+		                        config->boolValues[i] = true;
+				}
+				if (strcasecmp(ptr, "1")==0) {
```

Comment:
This was a very nitpicky comment... either `strcmp` or `strcasecmp` can be used on this line, so let's leave it.